application/src/main/java/org/thingsboard/server/controller/EntityProfileController.java

```diff
@@ -0,0 +1,127 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.controller;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+import org.thingsboard.server.common.data.EntityType;
+import org.thingsboard.server.common.data.audit.ActionType;
+import org.thingsboard.server.common.data.entityprofile.EntityProfile;
+import org.thingsboard.server.common.data.exception.ThingsboardException;
+import org.thingsboard.server.common.data.id.EntityProfileId;
+import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.common.data.page.PageData;
+import org.thingsboard.server.common.data.page.PageLink;
+import org.thingsboard.server.dao.entityprofile.EntityProfileService;
+import org.thingsboard.server.queue.util.TbCoreComponent;
+import org.thingsboard.server.service.security.model.SecurityUser;
+import org.thingsboard.server.service.security.permission.Operation;
+import org.thingsboard.server.service.security.permission.Resource;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+@RestController
+@TbCoreComponent
+@RequestMapping("/api")
+@Slf4j
+@RequiredArgsConstructor
+public class EntityProfileController extends BaseController {
+    private static final String ENTITY_PROFILES_ID = "entityProfilesId";
+    private final EntityProfileService service;
+
+    @PreAuthorize("hasAnyAuthority('SYS_ADMIN', 'TENANT_ADMIN')")
+    @GetMapping("/entityProfiles/{entityProfilesId}")
+    public EntityProfile getEntityProfileById(@PathVariable(ENTITY_PROFILES_ID) String strId) throws ThingsboardException {
+        checkParameter(ENTITY_PROFILES_ID, strId);
+        EntityProfileId id = new EntityProfileId(toUUID(strId));
+        SecurityUser user = getCurrentUser();
+        try {
+            EntityProfile entityProfile = service.findById(user.getTenantId(), id);
+            checkNotNull(entityProfile);
+            accessControlService.checkPermission(user, Resource.ENTITY_PROFILE, Operation.READ, id, entityProfile);
+            return entityProfile;
+        } catch (Exception e) {
+            throw handleException(e);
+        }
+    }
+
+    @PreAuthorize("hasAnyAuthority('SYS_ADMIN', 'TENANT_ADMIN')")
+    @PostMapping("/entityProfiles")
+    public EntityProfile saveEntityProfile(@RequestBody EntityProfile entityProfile) throws ThingsboardException {
+        entityProfile = entityProfile.toBuilder()
+                .tenantId(getCurrentUser().getTenantId())
+                .build();
+        Operation operation = entityProfile.getId() == null ? Operation.CREATE : Operation.WRITE;
```

Comment:
There is a new method called checkEntity. Please use it instead of checkPermission.

Reply:
done, but as for me it seems not good way to check some entity by calling base class method with all logic inside
BaseController.class already has 700+ lines and this number will be grown